starlette/routing.py

```diff
@@ -639,21 +600,31 @@ def url_path_for(self, name: str, **path_params: typing.Any) -> URLPath:
                 pass
         raise NoMatchFound(name, path_params)
 
-    async def startup(self) -> None:
+    async def startup(
+        self, state: typing.Optional[typing.Dict[str, typing.Any]]
+    ) -> None:
         """
         Run any `.on_startup` event handlers.
         """
         for handler in self.on_startup:
+            sig = inspect.signature(handler)
+            if len(sig.parameters) == 1 and state is not None:
```

Comment:
If the state is `None`, it means that the server doesn't support it, so we'll maintain the same error message.